src/main/java/org/springframework/retry/policy/ExpressionRetryPolicy.java

```diff
@@ -114,4 +117,28 @@ public boolean canRetry(RetryContext context) {
 		}
 	}
 
+	/**
+	 * Check if the expression is a template
+	 * @param expression the expression string
+	 * @return true if the expression string is a template
+	 * @deprecated in favor of using literal expression
+	 */
+	@Deprecated
+	private boolean isTemplate(String expression) {
+		return expression.startsWith(PARSER_CONTEXT.getExpressionPrefix()) && expression.endsWith(PARSER_CONTEXT.getExpressionSuffix());
```

Comment:
This is not correct. The expression can be like this: `foo#{bar.value}.baz`. With meaning to evaluate a template part and concatenate it with the rest as strings.
See `ParserContext` JavaDocs.

Also that doesn't make sense to deprecate private method.

The point was to log `warn` that templating is deprecated. Nothing more we can do from the context of variable.

Reply:
fixed!